cpp/include/utilities/cython.hpp

```diff
@@ -246,6 +301,29 @@ void call_sssp(raft::handle_t const& handle,
                vertex_t* predecessors,
                const vertex_t source_vertex);
 
+// wrapper for shuffling:
+//
+template <typename vertex_t, typename edge_t, typename weight_t>
+major_minor_weights_t<vertex_t, weight_t> call_shuffle(
+  raft::handle_t const& handle,
+  vertex_t* edgelist_major_vertices /* [IN] */,  // make_zip_iterator does not accept const !
```

Comment:
I have used make_zip_iterator with const.  Following function with simple thrust device vectors (adapted from the make_zip_iterator documentation) compiles and generates correct results for me.

```
void my_copy(int const *d_int_in,
              float const *d_float_in,
              int *d_int_out,
              float *d_float_out,
              int len) {
  thrust::copy(thrust::device,
               thrust::make_zip_iterator(thrust::make_tuple(d_int_in, d_float_in)),
               thrust::make_zip_iterator(thrust::make_tuple(d_int_in, d_float_in)) + len,
               thrust::make_zip_iterator(thrust::make_tuple(d_int_out, d_float_out)));
}
```

Reply:
Well, I'm getting compiler errors with const pointers.

Reply:
Perhaps there's something else that's the root cause of this.  It feels like we should tease that out, passing in const pointers here seems like the right thing to do.

Reply:
What seems to be happening is that `sort_and_shuffle()` assumes a zip iterator of non-const's. Which explains the compiler error. All other invocations of `sort_and_shuffle()` use zip iterators made of non const iterators / buffers.

Can we resolve this?

Reply:
BTW, the explanation for the non-const pre-condition is `sort_and_shuffle_value()`'s (eventual) call, through `sort_and_count()`, to `thrust::sort()`, which is in-place, hence requiring non-const iterators.

Reply:
That makes more sense.  Can we either eliminate the comment or change the comment in this code (and the other place where it happens) to reflect that the values are not const because the arrays are modified in-place?

Reply:
Actually, wouldn't this be [IN, OUT] because it's modified in-place?

Reply:
Done.